runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ActiveWorkState.java

```diff
@@ -129,13 +135,29 @@ synchronized ActivateWorkResult activateWorkForKey(ShardedKey shardedKey, Work w
       return ActivateWorkResult.EXECUTE;
     }
 
-    // Ensure we don't already have this work token queued.
+    // Check to see if we have this work token queued.
+    // This set is for adding remove-able WorkItems if they exist in the workQueue. We add them to
+    // this set since a ConcurrentModificationException will be thrown if we modify the workQueue
+    // and then resume iteration.
+    Set<WorkId> queuedWorkToRemove = new HashSet<>();
```

Comment:
instead of a set and then removeif, how about doing in a single pass with an iterator and Iterator.remove()

Reply:
done